plugins/modules/homebrew_cask.py

```diff
@@ -438,9 +435,13 @@ def _get_brew_version(self):
 
         rc, out, err = self.module.run_command(cmd, check_rc=True)
 
-        # get version string from first line of "brew --version" output
-        version = out.split('\n')[0].split(' ')[1]
-        self.brew_version = version
+        if rc != 0:
+            self.module.fail_json(msg="Failed to get brew version", stdout=out, stderr=err)
+        pattern = r"Homebrew (.*)(\d+\.\d+\.\d+)(-dirty)?"
+        rematch = re.search(pattern, out)
+        if not rematch:
+            self.module.fail_json(msg="Failed to match regex to get brew version")
```

Comment:
How about including the output in the error?
```suggestion
            self.module.fail_json(msg="Failed to match regex to get brew version", stdout=out)
```